github/src/main/java/org/arquillian/extension/governor/github/impl/reporter/GitHubGovernorRecorder.java

```diff
@@ -0,0 +1,115 @@
+import org.jboss.arquillian.test.spi.TestClass;
+import org.jboss.arquillian.test.spi.event.suite.After;
+import java.lang.reflect.Method;
+import java.util.TreeSet;
+
+
+/**
+ * @author <a href="mailto:dpawar@redhat.com">Dipak Pawar</a>
+ */
+
+public class GitHubGovernorRecorder {
+
+    @Inject
+    private Instance<GitHubGovernorConfiguration> gitHubGovernorConfigurationInstance;
+
+    @Inject
+    private Event<PropertyReportEvent> propertyReportEvent;
+
+    public void setGitHubGovernorConfigurationInstance(Instance<GitHubGovernorConfiguration> gitHubGovernorConfigurationInstance) {
+        this.gitHubGovernorConfigurationInstance = gitHubGovernorConfigurationInstance;
+    }
+
+    public void setPropertyReportEvent(Event<PropertyReportEvent> propertyReportEvent) {
+        this.propertyReportEvent = propertyReportEvent;
+    }
+
+    public void gitHubReportEntries(@Observes After event) {
+        final Method testMethod = event.getTestMethod();
+        final TestClass testClass = event.getTestClass();
+
+        final GitHub gitHubValue = getGitHubValue(testMethod, testClass);
+
+        if (gitHubValue != null) {
+            final GitHubGovernorConfiguration configuration = gitHubGovernorConfigurationInstance.get();
+
+            final String gitHubIssueURL = contructGitHubIssueURL(configuration, gitHubValue);
+
+            final Class<? extends Detectable>[] detectables = gitHubValue.detector().value();
+            final TreeSet<String> uniqDetectables = new TreeSet<String>();
+
+            for (final Class<? extends Detectable> detectable : detectables) {
+                uniqDetectables.add(detectable.getSimpleName());
+            }
+
+            String detectablesName = "";
+            for (final String detectable : uniqDetectables) {
+                detectablesName += (detectablesName == "" ? "" : ",") + detectable;
+            }
+
+            final TableEntry gitHubDetector = new TableEntry();
+            gitHubDetector.setTableName("GitHubOptions");
+            gitHubDetector.getTableHead().getRow().addCells(new TableCellEntry("Force"), new TableCellEntry("Detector Value"), new TableCellEntry("Detector Strategy"));
+
+            final TableRowEntry row = new TableRowEntry();
+            row.addCells(new TableCellEntry(String.valueOf(gitHubValue.force())), new TableCellEntry(String.valueOf(gitHubValue.detector().strategy().getSimpleName())), new TableCellEntry(detectablesName));
+            gitHubDetector.getTableBody().addRow(row);
+
+            propertyReportEvent.fire(new PropertyReportEvent(new KeyValueEntry("GitHub URL", gitHubIssueURL)));
+            propertyReportEvent.fire(new PropertyReportEvent(gitHubDetector));
+        }
+    }
+
+    private String contructGitHubIssueURL(GitHubGovernorConfiguration configuration, GitHub gitHub) {
+        String url = configuration.getDefaultGithubURL();
+        if (configuration.getRepositoryUser() != "") {
```

Comment:
ok, I am updating it with .isEmpty? method. just wondering do we have any disadvantages of this approach?